ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs

```diff
@@ -1940,6 +1942,44 @@ internal IEnumerable<ParameterDeclaration> MakeParameters(IReadOnlyList<IParamet
 			}
 		}
 
+		protected internal override TranslatedExpression VisitBlockContainer(BlockContainer container, TranslationContext context)
+		{
+			var oldReturnContainer = statementBuilder.currentReturnContainer;
+			var oldResultType = statementBuilder.currentResultType;
+			var oldIsIterator = statementBuilder.currentIsIterator;
+
+			statementBuilder.currentReturnContainer = container;
+			statementBuilder.currentResultType = context.TypeHint;
+			statementBuilder.currentIsIterator = false;
+			try {
+				var body = statementBuilder.ConvertAsBlock(container);
+				body.InsertChildAfter(null, new Comment(" Could not convert BlockContainer to single expression"), Roles.Comment);
+				var ame = new AnonymousMethodExpression();
+				ame.Body = body;
+				var inferredReturnType = InferReturnType(body);
+				var func = compilation.FindType(typeof(Func<>)).GetDefinition();
+				var delegateType = new ParameterizedType(func, inferredReturnType);
```

Comment:
This will crash if `GetDefinition()` returns null.

Reply:
Fixed